xdsl/parser/core.py

```diff
@@ -856,7 +856,7 @@ def _parse_generic_operation(self, op_type: type[Operation]) -> Operation:
             successors = []
 
         # Parse attribute dictionary
-        properties = self.parse_optional_properties_dict()
+        properties = self.parse_optional_properties_dict() or {}
```

Comment:
I know this is pedantic but it avoids an extra empty dict creation
```suggestion
        properties = self.parse_optional_properties_dict()
        if properties is None:
            properties = {}
```